### ISSUE TITLE
Implementing a prim type in a class expression should report error

### ISSUE THREAD

**Issue by bvanjoi** [microsoft/TypeScript#60484]
### 🔎 Search Terms

class expression implement primitive type

### 🕗 Version & Regression Information

- This changed between versions __5.2.2____ and __5.3.3_____

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.3.3#code/MYewdgzgLgBAgjAvDYAbAhhCMBCMCWAtgA6oCmhZYU2YAroQEZkBOMA3gL4BQ3am2AMIES5StVoNmbLkA

### 💻 Code

```ts
class C implements number {} // report that class cannot implement primitive type

const A = class B implements number {} // this code should report the same error, but it actually does not.
```

### 🙁 Actual behavior

Refer to the comment above

### 🙂 Expected behavior

Refer to the comment above

**Comment by jcalz** [comment]
Looks like #55712
